presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergConnector.java

```diff
@@ -0,0 +1,200 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.iceberg;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.bootstrap.LifeCycleManager;
+import io.airlift.log.Logger;
+import io.prestosql.plugin.hive.HiveTransactionHandle;
+import io.prestosql.spi.classloader.ThreadContextClassLoader;
+import io.prestosql.spi.connector.Connector;
+import io.prestosql.spi.connector.ConnectorAccessControl;
+import io.prestosql.spi.connector.ConnectorCapabilities;
+import io.prestosql.spi.connector.ConnectorMetadata;
+import io.prestosql.spi.connector.ConnectorNodePartitioningProvider;
+import io.prestosql.spi.connector.ConnectorPageSinkProvider;
+import io.prestosql.spi.connector.ConnectorPageSourceProvider;
+import io.prestosql.spi.connector.ConnectorSplitManager;
+import io.prestosql.spi.connector.ConnectorTransactionHandle;
+import io.prestosql.spi.connector.SystemTable;
+import io.prestosql.spi.connector.classloader.ClassLoaderSafeConnectorMetadata;
+import io.prestosql.spi.session.PropertyMetadata;
+import io.prestosql.spi.transaction.IsolationLevel;
+
+import java.util.List;
+import java.util.Set;
+
+import static com.google.common.collect.Sets.immutableEnumSet;
+import static io.prestosql.spi.connector.ConnectorCapabilities.NOT_NULL_COLUMN_CONSTRAINT;
+import static io.prestosql.spi.transaction.IsolationLevel.READ_COMMITTED;
+import static io.prestosql.spi.transaction.IsolationLevel.checkConnectorSupports;
+import static java.util.Objects.requireNonNull;
+
+public class IcebergConnector
+        implements Connector
+{
+    private static final Logger log = Logger.get(IcebergConnector.class);
+
+    private final LifeCycleManager lifeCycleManager;
+    private final IcebergTransactionManager transactionManager;
+    private final IcebergMetadataFactory metadataFactory;
+    private final ConnectorSplitManager splitManager;
+    private final ConnectorPageSourceProvider pageSourceProvider;
+    private final ConnectorPageSinkProvider pageSinkProvider;
+    private final ConnectorNodePartitioningProvider nodePartitioningProvider;
+    private final Set<SystemTable> systemTables;
+    private final List<PropertyMetadata<?>> sessionProperties;
+    private final List<PropertyMetadata<?>> schemaProperties;
+    private final List<PropertyMetadata<?>> tableProperties;
+    private final ConnectorAccessControl accessControl;
+
+    public IcebergConnector(
+            LifeCycleManager lifeCycleManager,
```

Comment:
This is annoying since `stop()` currently throws a checked exception. I'll update all the connectors to use `Runnable` after we upgrade Airlift with the lifecycle changes.